bot/exts/moderation/voice_gate.py

```diff
@@ -160,6 +187,47 @@ async def on_message(self, message: discord.Message) -> None:
         with suppress(discord.NotFound):
             await message.delete()
 
+    @Cog.listener()
+    async def on_voice_state_update(
+            self,
+            member: Member,
+            before: discord.VoiceState,
+            after: discord.VoiceState
+    ) -> None:
+        """Pings a user if they've never joined the voice chat before and aren't voice verified."""
+        if member.bot:
+            log.trace("User is a bot. Ignore.")
+            return
+
+        # member.voice will return None if the user is not in a voice channel
+        if member.voice is None:
+            log.trace("User not in a voice channel. Ignore.")
+            return
+
+        in_cache = await self.redis_cache.get(member.id, NO_MSG)
+        if in_cache:
+            log.trace("User already in cache. Ignore.")
+            return
+
+        log.trace("User not in cache and is in a voice channel.")
+        verified = any(Roles.voice_verified == role.id for role in member.roles)
+        if verified:
+            log.trace("User is verified, add to the cache and ignore.")
+            await self.redis_cache.set(member.id, NO_MSG)
+            return
+
+        log.trace("User is unverified. Send ping.")
+        await self.bot.wait_until_guild_available()
+        voice_verification_channel = self.bot.get_channel(Channels.voice_gate)
+
+        message = await voice_verification_channel.send(f"Hello, {member.mention}! {VOICE_PING}")
+        await self.redis_cache.set(member.id, message.id)
+
+        await asyncio.sleep(60)
+        if message := await self.redis_cache.get(member.id):
+            await message.delete()
+            await self.redis_cache.set(member.id, NO_MSG)
```

Comment:
This should still use the configured value, not a hardcoded `60`. For safety it can still suppress `NotFound` on the delete, although it should no longer be possible unless we get very unlucky and the `on_message` runs in parallel (the cache access is not synchronous).

Reply:
Use config in d6820a2, correctly delete in 92132af.